# U.S. Medical Insurance Costs

In this project I will be analyzing the insurance costs of smokers compared to non-smokers from a **CSV** file that includes these six different factors:
* _Age_
* _Sex_
* _Bmi_
* _Number of children_
* _Whether or not the person is a smoker_
* _The region that the person is from_

But I will not be needing most of them as I will just be using the _**smoker**_ catagory

First I will import the dataset for analysis and get rid of other data that I will not need

In [1]:
import csv

with open('insurance.csv') as insurance_csv:
    insurance_reader = csv.DictReader(insurance_csv)
    insurance_list = [row for row in insurance_reader]

# This gets rid of the data I don't need
smoker_list = [{'smoker': person['smoker'], 'cost': person['charges']} for person in insurance_list]

# This converts the word 'Yes' into a 1 and the word 'No' into a 0
for smoker in smoker_list:
    if smoker['smoker'] == 'yes':
        smoker['smoker'] = 1
    else:
        smoker['smoker'] = 0



Now I will write a function that will return the average cost for a list of people given if they are a smoker or not.

In [2]:
def average_cost(people, smoker):
    total = 0
    count = 0
    for person in people:
        if person['smoker'] == smoker:
            total += float(person['cost'])
            count += 1
    return total / count

Now I'll:
1. Get the average cost for the _**non-smokers**_
2. Get the average cost for the _**smokers**_
3. Print out the difference between them

In [3]:
# Non-smoker avg:
non_smoker_avg = average_cost(smoker_list, 0)
# Smoker avg:
smoker_avg = average_cost(smoker_list, 1)
smoker_avg_diff = smoker_avg - non_smoker_avg


'The non-smoker average cost is {} and the smoker average cost is {} and the average difference between the non-smokers and the smokers is: {}'.format(non_smoker_avg, smoker_avg, smoker_avg_diff)

'The non-smoker average cost is 8434.268297856199 and the smoker average cost is 32050.23183153285 and the average difference between the non-smokers and the smokers is: 23615.96353367665'

So this concludes that the insurance costs for these individuals is on average $23,615.96 cheaper than smokers!